### ISSUE TITLE
Live preview empty when using `:backend: slidy`

### ISSUE THREAD

**Issue by arnauldvm** [asciidoctor/asciidoctor-vscode#198]
## Description

Preview window is empty when using `:backend: slidy`

Even though slidy backend might not be supported, shouldn't the preview fallback to a defaut backend in that case?

## System Information

```
Version: 1.33.1 (system setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:09.080Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT ia32 10.0.17134
```

```
AsciiDoc
joaompinto.asciidoctor-vscode
João Pinto
v2.7.2
```

## To Reproduce

1. Document containing line `:backend: slidy`
2. Preview window is empty.
3. Comment the line `:backend: slidy`
4. Preview window is not empty

## Screenshots & Files

N/A

## Additional Context

I also see many errors in the log, but they do not seem to be related to the error.
(They also appear when no backend is specified.)

```text
[2019-05-12 10:24:32.024] [exthost] [error] [joaompinto.asciidoctor-vscode] provider FAILED
[2019-05-12 10:24:32.024] [exthost] [error] TypeError: tokens.filter is not a function
    at AsciidocFoldingProvider.<anonymous> (C:\Users\avm\.vscode\extensions\joaompinto.asciidoctor-vscode-2.7.2\src\features\foldingProvider.ts:28:32)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\avm\.vscode\extensions\joaompinto.asciidoctor-vscode-2.7.2\out\src\features\foldingProvider.js:8:58)
```

**Comment by arnauldvm** [comment]
Found this workaround :

Add this to `settings.json`:

```json
    "asciidoc.preview.attributes": {
        "backend": "html5",
    },
```

**Comment by MatteoCampinoti94** [comment]
Alas slidy is not available as a backend for asciidoctor, you need to use asciidoc for it to work :(
Your workaround worked because by using `"backend": "html5"` instead of `"backend": "html5@"` you are overriding the backend attribute in the document, so "slidy" never reaches the converter and no error occurs. The `asciidoc.preview.attributes` setting follows the same attribute rules as the command line version:
* `"attribute": "value"` set attribute and override
* `"attribute": "value@"` set attribute and do not override
* `"attribute!": "` unset attribute and override

As for the blank preview pane: when the asciidoctor.js converter fails the pane simply has no data to display. However even if we could catch a backend error (which we cannot as it throws a general `NotImplementedError` exception) we wouldn't change it to default, the preview would not reflect the contents of the source document.
I added an error notification to warn the user of any error occurring during the conversion, so even if the preview is empty you will be able to tell what the error is. It will be available in version 2.7.3 very soon.

The FoldingProvider error is know, it is not causing any problems to the extension beyond the error message though. We are in the process of overhauling our code base, which will include folding, so we prefer not to devote time to fixing it at the moment.

Hope this helped you :)

And thank you for using the issue template! :heart:
